contracts/domainRegistry.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "./domainsSetLibrary.sol";
+
+contract DomainRegistry {
+    uint256 public collateral = 0.001 ether;
+    using domainSet for domainSet.Set;
+
+    domainSet.Set internal domains;
+
+    event DomainRegistered(
+        string domainName,
+        address indexed owner,
+        uint256 deposit,
+        string operation
+    );
+
+    function registerDomain(string memory domainName) external payable {
+        require(msg.value == collateral, "Deposit must equal to collateral");
+        domains.add(domainName);
+        emit DomainRegistered(domainName, msg.sender, msg.value, "create");
```

Comment:
It should be less gas efficient to save the same parameter "create" or "remove" than just set two different events like `DomainRegistered`, `DomainReleased`
       -1 point